test/export/MappingExporter.test.ts

```diff
@@ -194,6 +194,116 @@ describe('MappingExporter', () => {
+
+      exporter.export();
+      expect(loggerSpy.getAllMessages('error')).toHaveLength(0);
+      expect(observation.mapping.length).toBe(originalMappingLength); // No metadata added
+      expect(status.mapping.length).toBe(originalStatusMappingLength + 1); // New rule added to status element
+    });
+
+    it('should not log an error and not add metadata but add rules for a Mapping that is inherited from the parent with the same metdata', () => {
+      /**
+       * Mapping: rim
+       * Id: rim
+       * Source: MyObservation
+       * Title: "RIM Mapping"
+       * Target: "http://hl7.org/v3"
+       * * status -> "Something.new"
+       */
+
+      const mapping = new Mapping('rim');
+      mapping.source = 'MyObservation';
+      mapping.id = 'rim';
+      mapping.title = 'RIM Mapping';
+      mapping.target = 'http://hl7.org/v3';
+      const newRule = new MappingRule('status');
+      newRule.map = 'Something.new';
+      mapping.rules.push(newRule);
+      doc.mappings.set(mapping.name, mapping);
+
+      const originalMappingLength = observation.mapping.length;
+      const status = observation.elements.find(e => e.id === 'Observation.status');
+      const originalStatusMappingLength = status.mapping.length;
+
+      exporter.export();
+      expect(loggerSpy.getAllMessages('error')).toHaveLength(0);
+      expect(observation.mapping.length).toBe(originalMappingLength); // No metadata added
+      expect(status.mapping.length).toBe(originalStatusMappingLength + 1); // New rule added to status element
+    });
+
+    it('should log an error and not add mapping or rules when a Mapping has the same identity as one on the parent but has other metadata that differs', () => {
+      /**
+       * Mapping: rim
+       * Id: rim
+       * Source: MyObservation
+       * Title: "RIM Mapping"
+       * Target: "http://real.org/not"
+       * * status -> "Something.new"
+       */
+
+      const mapping = new Mapping('rim');
+      mapping.source = 'MyObservation';
+      mapping.id = 'rim';
+      mapping.title = 'RIM Mapping';
+      mapping.target = 'http://real.org/not';
+      const newRule = new MappingRule('status');
+      newRule.map = 'Something.new';
+      mapping.rules.push(newRule);
+      doc.mappings.set(mapping.name, mapping);
+
+      const originalMappingLength = observation.mapping.length;
+      const status = observation.elements.find(e => e.id === 'Observation.status');
+      const originalStatusMappingLength = status.mapping.length;
+
+      exporter.export();
+      expect(loggerSpy.getAllMessages('error')).toHaveLength(1);
+      expect(observation.mapping.length).toBe(originalMappingLength); // No metadata added
+      expect(status.mapping.length).toBe(originalStatusMappingLength); // No rule added to status element
+    });
+
+    it('should not log an error and not add metadata but add rules for a Mapping that is inherited from the parent and has additional metadata not on the parent', () => {
```

Comment:
I feel like this test description is a little confusing.  This test is about testing that changing the description doesn't log an error, right?  And it should probably also verify that the new description _does_ get set in the FHIR mapping?